src/tilesource.js

```diff
@@ -276,11 +291,19 @@ $.TileSource.prototype = /** @lends OpenSeadragon.TileSource.prototype */{
     getClosestLevel: function( rect ) {
         var i,
             tilesPerSide,
-            tiles;
+            tiles,
+            tileSize;
+
         for( i = this.minLevel; i < this.maxLevel; i++ ){
             tiles = this.getNumTiles( i );
-            tilesPerSide = Math.floor( Math.max( rect.x, rect.y ) / this.getTileSize(i) );
-            if( Math.max( tiles.x, tiles.y ) + 1 >= tilesPerSide ){
+            tileSize = this.getTileSize(i);
+
+            tilesPerSide = new $.Point(
+              Math.floor( rect.x / tileSize.x ),
+              Math.floor( rect.y / tileSize.y )
+            );
+            
+            if( tiles.x + 1 >= tilesPerSide.x && tiles.y + 1 >= tilesPerSide.y ){
```

Comment:
I may be misreading, but it seems a more accurate translation of the `Math.max` from the original would be to have `||` rather than `&&` here.